### PR TITLE
Add 4.0-dev branch to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by PhilETaylor**
Add 4.0-dev branch to CONTRIBUTING.md

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -23,3 +23,4 @@ There are other branches available which serve specific purposes.
 | master | Each commit made to staging gets tested if it passes unit tests and codestyle rules. It is then merged into master. This is done automatically. |
 | 2.5.x | Branch for the Joomla 2.5.x series. Support for this version has ended, no patches are accepted here. |
 | 3.6.x | Branch for the next minor Joomla version. New backward compatible features go into this branch. Commits to staging will be applied to this branch as well. |
+| 4.0-dev | Branch for the next major Joomla version. New backward in-compatible features go into this branch. Commits to staging will be applied to this branch as well. |
```
Comment (brianteeman) [inline .github/CONTRIBUTING.md]: No need to hyphenate incompatible it's a real word

Comment (PhilETaylor) [reply .github/CONTRIBUTING.md]: Done.

**Review by PhilETaylor — COMMENTED** [review]
Done.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on cb025abcb8cd000bdecf6c137591699fe759bb75<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12774">issues.joomla.org/tracker/joomla-cms/12774</a>.</sub>

**Comment by jeckodevelopment** [comment]
I have tested this item :white_check_mark: successfully on cb025abcb8cd000bdecf6c137591699fe759bb75<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12774">issues.joomla.org/tracker/joomla-cms/12774</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12774">issues.joomla.org/tracker/joomla-cms/12774</a>.</sub>

**Comment by Bakual** [comment]
Merged. Looking at that file it needs some other updates as well. Eg the 2.5.x and 3.6.x branches don't exist anymore. Or maybe we can rewrite it to be more generic?

**Comment by PhilETaylor** [comment]
I think a complete explanation of each branch is required, and not a generic file. 

I dont want to add the 3.7.x line as I honestly have lost track of whats going into that :) :-)

**Comment by Bakual** [comment]
>I dont want to add the 3.7.x line as I honestly have lost track of whats going into that

It's been merged into staging and is awaiting it's termination. 😄

**Pull request merged**